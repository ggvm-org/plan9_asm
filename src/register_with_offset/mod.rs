use std::fmt;

pub mod register;

pub use register::Register;

#[derive(Debug, PartialEq)]
pub struct RegisterWithOffset {
    pub offset: usize,
    pub register: Register,
}

impl fmt::Display for RegisterWithOffset {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let s = {
            let offset = self.offset;
            let register = self.register;
            if offset == 0 {
                register.to_string()
            } else {
                format!("{offset}({register})")
            }
        };
        write!(f, "{s}")
    }
}

impl From<Register> for RegisterWithOffset {
    fn from(register: Register) -> Self {
        RegisterWithOffset {
            offset: 0,
            register,
        }
    }
}
