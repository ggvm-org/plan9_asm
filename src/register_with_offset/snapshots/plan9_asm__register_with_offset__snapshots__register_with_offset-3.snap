---
source: src/register_with_offset/mod.rs
assertion_line: 70
expression: register_with_offset!(TEST_OFFSET => SP)

---
10(SP)
